apps/music/js/metadata/flac.js

```diff
@@ -82,31 +110,31 @@ var FLACMetadata = (function() {
    *       header).
    *     {BlobView} view The BlobView with the block's data, starting at the
    *       beginning of the metadata block's content (not the header).
+   *   The second argument is an error if any.
    */
-  function readMetadataBlockHeader(blobview) {
-    return new Promise(function(resolve, reject) {
-      var header = blobview.readUnsignedByte();
-      var last = (header & 0x80) === 0x80;
-      var block_type = header & 0x7F;
-      var length = blobview.readUint24(false);
-
-      // Get the contents of this block, plus enough to read the next block's
-      // header if necessary.
-      blobview.getMore(blobview.viewOffset + blobview.index, length + 4,
-                       function(more, err) {
+  function readMetadataBlockHeader(blobview, callback) {
```

Comment:
I think we should leave this function how it was; turning it into a callback doesn't simplify things here.